src/main/kotlin/me/zeroeightsix/kami/feature/module/Nuker.kt

```diff
@@ -0,0 +1,132 @@
+package me.zeroeightsix.kami.feature.module
+
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+import me.zero.alpine.listener.EventHandler
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.event.TickEvent
+import me.zeroeightsix.kami.util.asVec3d
+import me.zeroeightsix.kami.util.single_vec
+import net.minecraft.block.CommandBlock
+import net.minecraft.block.FluidBlock
+import net.minecraft.block.JigsawBlock
+import net.minecraft.block.StructureBlock
+import net.minecraft.client.network.ClientPlayerEntity
+import net.minecraft.client.world.ClientWorld
+import net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Box
+import net.minecraft.util.math.Direction
+import net.minecraft.util.math.Vec3d
+import net.minecraft.world.RaycastContext
+
+@Module.Info(name = "Nuker", category = Module.Category.PLAYER, description = "Destroys Blocks around the Player")
+object Nuker : Module() {
+    @Setting(name = "Hit Range")
+    private var hitRange: @Setting.Constrain.Range(min = 0.0, max = 8.0, step = 1.0) Double = 4.0
+
+    @Setting(name = "Through Blocks")
+    private var throughBlocks = false
+
+    @Setting(name = "Select Tool", comment = "select the best tool for the block being broken")
+    private var selectTool = false
+
+    @EventHandler
+    private val updateListener = Listener<TickEvent.InGame>({
+        instantMineBlocks(it.player, it.world)
+
+        if (currentBlock == null) {
+            nextBlock(it.player, it.world)
```

Comment:
The function name looks like it should return a value, can you make it so that

  - Its name does not suggest a return value (findNextBlock)
  - make it return a value, so you would not access the field from the listener directly.

Reply:
To add to this, it is cleaner to let kotlin infer `currentBlock` being nonnull. Let `nextBlock` return the next block instead of assigning in the function, and then assign it to `currentBlock` like so:
```kotlin
currentBlock = nextBlock(it.player, it.world) ?: return@Listener
```
kotlin _should_ be able to tell that `currentBlock` is never null due to the return statement, so the `!= null` check in the next if statement isn't needed. (smart casting)

Reply:
you will need to declare it as a local variable in this case, because kotlin thinks that another thread can be modifying this field. You would use a local variable and have the `nextBlock` function modify the field in this case.